simulator/infra/ports/module.h

```diff
@@ -42,6 +42,11 @@ class Module : public Log
     void enable_logging_impl( const std::unordered_set<std::string>& names);
     boost::property_tree::ptree topology_dumping_impl() const;
 
+    void enable_prefetch(bool key, uint32 fetchahead_size, std::string prefetch_method); // enable-prefetch function initialization
+    bool prefetch_enabled = false; // boolean value that represents that prefetch is enabled/disabled
+    uint32 fetchahead_size; // value of fetchahead distance size
+    std::string prefetch_method; // value of prefetch method
```

Comment:
The problem is that _all_ modules have knowledge regarding the prefetching. That's why we are trying to keep features locally.